generate/sre-authorization/osd-sre-admin.sh

```diff
@@ -0,0 +1,156 @@
+#!/bin/bash
+
+BLACKLIST=`cat blacklist.json`
+
+echo "apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: osd-sre-admin
+rules:"
+
+# /api
+GROUP=""
+for VERSION in `oc get --raw /api | jq -r .versions[] | sort`;
```

Comment:
We are going to be running this rebuild script periodically and I think we want to avoid the dynamic rebuild and, instead, provide a static input file to the script. The inputs can be the safelist and blocklist of resources the group are allowed to access, and not allowed to access, respectively.

What leads me to the conclusion is the pre-run setup: log into a cluster (where are the creds? how to keep them safe?), connect to it (what if there's a network partition?) and fetch the resources. This is a kind of toil and could lead to future issues. In addition to my setup concern, I have a readability concern as well. 

We don't know what the output of the script will be by only reading it because the inputs are essentially dynamic (via the `oc` command). If we provide the resources in plain text then we can better predict what happens when it runs.

Reply:
But that's part of the problem, the inputs are dynamic.  If we have 10 new CR's in an OSD cluster that we need to manage, we need to add them to the osd-sre-admin clusterrole.  

I do not see this script being run often.  The clusterrole could be managed manually if it's better.  But I think we should keep it around until a bit past GA so we can easily generate something new.

On knowing what will change, that's true.  But we have a baseline to look at for future use of the script, simply `make; git diff` and see what changed.  The output of the script is static.  CI should not run the script.  So CI will only ever see a static file.  Humans need to review changes to ensure the delta is good.